src/equilibrate_utils.jl

```diff
@@ -0,0 +1,726 @@
+
+#--- Fixes parameters, required when some exponents contain parameters. --#
+#Calls the internal make_polynomial function to create a polynomial. Used to model creation (will succced unless there is parameters in the exponents). Input include list of parameters to fix. All parameters occuring as exponent must be here (none other is required at this stage). Will insert fixed concentrations if such exists.
+"""
+    fix_parameters(reaction_network, [parameter_values]; t=get_polyvars(rn).t, kwargs...)
+
+Fixes certain parameters of a reaction network, preparing it for homotopy continuation.
+
+In the polynomial needed to be solved by homotopy continuation to find steady states solutions, both varriables and parameters are initially represented as polynomial varriables. If a parameter exists in an exponent (e.g. X^n), that parameter might be needed to be fixed to a desired value before solving. Fixing is not required, but if you solve a polynomial several times, it might increase performance.
+
+
+## args
+-  reaction_network: a reaction network.
+-  parameter_values: An entire vector can be given, fixing all parameters. Generaly you do not want to do this.
+-  t=get_polyvars(rn).t: If you want to fix the time value.
+
+## kwargs
+-  How parameters generally are fixed. Write e.g. n=4 to fix the parameter value of n.
+"""
+function fix_parameters(rn::DiffEqBase.AbstractReactionNetwork, params=Vector{Number}()::Vector{Number}; t=get_polyvars(rn).t, full_vector_exemption=-1, kwargs...)
+    check_is_polynomial(rn)
+    set_equi_poly!(rn,t;internal___var___paramvals=params,internal___var___paramvals___exemption=full_vector_exemption,kwargs...)
+    add_constraints!(rn)
+    return nothing
+end
+
+#--- Adds additional constraints to the system, typically a fixed concentration such as in the system X <--> Y.
+#In some networks some linear combinations of concentrations remain fixed. This supplies this information directly to the network.
+"""
+    @add_contraint(reaction_network, constraints)
+
+Some networks (like X ↔ Y) have an infinite number of potential steady states, depending on initial conditions. For these some additional contraint(s) is required. Typically in the form of fixed concentrations.
+
+## args
+-  reaction_network: a reaction network.
+-  constraints: Constraints, typically given as X + Y = C. Both parameters, varriables, and numbers can be given. However, parameters much be declared previously in the reaction network.
+"""
+macro add_constraint(rn::Symbol, constraints::Expr...)
+    func_expr = Expr(:escape,:(internal___add___constraint!($rn)))
+    foreach(constraint -> push!(func_expr.args[1].args,recursive_replace_vars!(balance_poly(constraint), rn)),constraints)
+    return func_expr
+end
+#In some networks some linear combinations of concentrations remain fixed. This supplies this information directly to the network.
+"""
+    @add_contraints(reaction_network, constraints)
```

Comment:
If we add constraints we should get a DAE, but I don't see a mass matrix being added to the `ODEProblem`?

Reply:
That is exactly what I had in mind when I suggested using `add_constraints` instead of `fix_concentrations` for the macro name. Right now constraints are only added into the equilibration data structures, but ultimately we should extend this to allow the generation of DAE models (or at least allow them when using min reaction networks where ODE construction is deferred).

Reply:
I have never really been involved with DAEs, but do we really get one? The simplest examplee would be X <--> Y, with the equations
dX/dt = Y - X
dY/dt = X - Y
Adding the information X+Y=1 doesn't really make the system different, since it is already defined by one differential equation for each variable, and an initial condition.

However, when we want to find steady states we get the system
Y - X = 0
X -Y = 0
which have an infinite amount of solutions, we need some additional information. Here a human "knows" the relation X+Y=1 (or = some parameter).`@add_constraint` adds this information, making the system
Y - X = 0
X -Y = 0
X+Y-1 =0
which we can solve. 

It is likely that one can, given some initial condition, compute this constraint for a reaction network. Adding such an algorithm would save one from having from defining the constraints (although one would still have to supply some initial condition, or name for a parameter designating the combined concentration).

All that said, adding a mass matrix to the ODE problem would still make sense (I presume this could increase efficiency?). Or would creating a DAE problem from it also be a way to solve the system, given this additional information, in a more efficient way?